Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/use-current-contest.tsx

```diff
@@ -215,118 +219,123 @@ const CurrentContestsProvider = ({ children }: ICurrentContestsProviderProps) =>
-                setContestError(startContestError);
+            if (!isNil(registerContestError)) {
+                setContestError(registerContestError);
+                setRequirePassword(false);
                 return;
             }
 
-            const {
-                contest: newContest,
-                contestIsCompete,
-                participantId: currentParticipantId,
-                remainingTimeInMilliseconds: newRemainingTimeInMilliseconds,
-                totalParticipantsCount: newTotalParticipants,
-                activeParticipantsCount: newActiveParticipants,
-            } = startContestData;
+            const { requirePassword: responseRequirePassword } = registerForContestData;
 
-            setContest(newContest);
-            setIsOfficial(contestIsCompete);
-            setParticipantId(currentParticipantId);
-            setRemainingTimeInMilliseconds(newRemainingTimeInMilliseconds);
-            setUserSubmissionsTimeLimit(startContestData.userSubmissionsTimeLimit);
-            setTotalParticipantsCount(newTotalParticipants);
-            setActiveParticipantsCount(newActiveParticipants);
+            setContest({ id: registerForContestData.id, name: registerForContestData.name } as IContestType);
+            setRequirePassword(responseRequirePassword);
         },
-        [ startContestData, startContestError ],
+        [ registerForContestData, registerContestError ],
     );
 
     useEffect(() => {
-        if (isNil(registerForContestParams)) {
-            return;
-        }
-
-        const { id } = registerForContestParams;
-        if (Number.isNaN(id)) {
+        if (isNil(submitContestPasswordUrlParams)) {
             return;
         }
 
         (async () => {
             startLoading();
-            await registerForContest();
+            await submitContestPassword({ Password: contestPassword });
             stopLoading();
         })();
-    }, [ registerForContest, registerForContestParams, startLoading, stopLoading ]);
+    }, [ contestPassword, submitContestPassword, submitContestPasswordUrlParams, startLoading, stopLoading ]);
 
     useEffect(
         () => {
-            if (isNil(registerForContestData)) {
+            if (isNil(submitContestPasswordResponse)) {
                 return;
             }
 
-            if (!isNil(registerContestError)) {
-                setContestError(registerContestError);
-                setRequirePassword(false);
+            if (!isNil(submitContestPasswordError)) {
+                setContestPasswordError(submitContestPasswordError);
+                setIsPasswordValid(false);
+
                 return;
             }
 
-            const { requirePassword: responseRequirePassword } = registerForContestData;
-
-            setContest({ id: registerForContestData.id, name: registerForContestData.name } as IContestType);
-            setRequirePassword(responseRequirePassword);
+            setIsPasswordValid(true);
```

Comment:
Make sure that this value is reset when making the request to register for contest again, otherwise we can deal with leftover state

Reply:
I am resetting it on row 276